### PR TITLE
Add "headerSchema"

### PR DESCRIPTION
**Pull request by handrews**
This addresses the request half of #296, and is a companion
keyword to "targetHints" (#383).

Like "targetHints", this is a rough first pass to get the
discussion of specifics started.  I expect it to be somewhat
underspecified to encourage experimentation.

I intentionally chose an example that is both very complex
and very useful.  It might also be worth putting up a simpler
example.  Although when this appears right after "targetHints",
the simpler header value examples in that section will help
motivate what is going on here.

### CODE REVIEW

File: jsonschema-hyperschema.xml [hunk 1]
```diff
@@ -1095,6 +1097,123 @@ GET /foo/
                 </section>
             </section>
 
+            <section title="headerSchema" anchor="headerSchema">
+                <t>
+                    <cref>
+                        As with "targetHints", this keyword is somewhat under-specified
+                        to encourage experiementation and feedback as we try to balance
```
[the remark this answers is not in the document]
Comment (handrews) [reply jsonschema-hyperschema.xml]: It's extra-experimental and experiential!

File: links.json [hunk 2]
```diff
@@ -38,6 +38,12 @@
         "mediaType": {
             "type": "string"
         },
+        "headerSchema": {
+            "description": "Schema for protocol-specific headers or analogous data, as signified by URI scheme",
```
Comment (dlax) [inline links.json]: Maybe drop "description" here as well?

Comment (handrews) [reply links.json]: srsly?  I swear I've pushed three times as many rebases/fixes as there are PRs tonight...

**Comment by handrews** [comment]
Last call- barring objections or reasonable requests for more time to review, this will get merged tomorrow.

**Pull request merged**